src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -159,5 +162,159 @@ public void HotReloadConfigRestRuntimeOptions()
                 File.Delete(configName);
             }
         }
+
+        #region ConfigFileWatcher NewFileContentsDetected event invocation tests
+
+        private const string UNEXPECTED_INVOCATION_COUNT_ERR = "Unexpected number of invocations of the NewFileContentsDetected event.";
+        private const string FILECHANGE_EVENT_NOT_RAISED_ERR = "The file system's file-changed event was not raised.";
+
+        /// <summary>
+        /// This test validates that overwriting a file with the different contents
+        /// results in ConfigFileWatcher's NewFileContentsDetected event being raised.
+        /// Internally, ConfigFileWatcher.NewFileContentsDetected is raised when the hash
+        /// of the file contents differ from what was previously detected. For this test,
+        /// the file contents differ MyValue -> MyChangedValue.
+        /// The config file name is specific to this test in order to avoid concurrently
+        /// running tests from stepping over each other due to acquiring a handle(s) to the file.
+        /// </summary>
+        [TestMethod]
+        public void ConfigFileWatcher_NotifiedOfOneNetNewChanges()
+        {
+            // Arrange
+            int fileChangeNotificationsReceived = 0;
+            string configName = "ConfigFileWatcher_NotifiedOfOneNetNewChanges.json";
+
+            // Mock filesystem calls for when FileUtilities.ComputeHash() utilizes the filesystem.
+            // For this test, we assume the file exists because we are validating the behavior
+            // of differing config file contents triggering a NewFileContentsDetected event.
+            IFileSystem fileSystem = Mock.Of<IFileSystem>();
+            Mock.Get(fileSystem).Setup(fs => fs.Directory.GetCurrentDirectory()).Returns(Directory.GetCurrentDirectory());
+            Mock.Get(fileSystem).Setup(fs => fs.File.Exists(It.IsAny<string>())).Returns(true);
+
+            // Mock file system to return different byte arrays for each read which occurs in
+            // FileUtilities.ComputeHash() called by ConfigFileWatcher.
+            // The first read occurs during ConfigFileWatcher's initialization
+            // The second read occurs during the manual invocation of a "file changed" event.
+            Mock.Get(fileSystem).SetupSequence(fs => fs.File.ReadAllBytes(It.IsAny<string>()))
+                .Returns(Encoding.UTF8.GetBytes("FirstValue"))
+                .Returns(Encoding.UTF8.GetBytes("SecondValue"))
+                .Returns(Encoding.UTF8.GetBytes("SecondValue"));
```

Comment:
IF we care only about 2 reads, why mock a 3rd read as well?

Reply:
We care about two reads as a result of the hot reload.

First Read: original configuration loading.
Second Read: First simulation of file system FileChangedEvent
Third Read: Result of 2nd simulation of file system FileChangedEvent because a real filesystem returns >1 filechanged event. This test validates that the second return of "SecondValue" is a no-op because we've already reacted to seeing "SecondValue"